Source/RunActivity/Viewer3D/Cameras.cs

```diff
@@ -1778,6 +1783,13 @@ public class TracksideCamera : LookAtCamera
         const float CameraAltitude = 2;
         // Height above the coordinate center of target.
         const float TargetAltitude = TerrainAltitudeMargin;
+        const int MaximumSpecialPointDistance = 300;
+        const float PlatformOffsetM = 3.3f;
+        public bool SpecialPoints = false;
+        protected bool SpecialPointFound = false;
+        const float CheckIntervalM = 50f; // every 50 meters it is checked wheter there is a near special point
+        protected float DistanceRunM = 0f; // distance run since last check interval
+        protected bool FirstUpdateLoop = true; // first update loop
```

Comment:
I don't think putting all of the special trackside camera fields into the base trackside camera class is a great way to organise this. There should be only tiny, or no, changes to `TracksideCamera` for this feature.

One possible way of doing that is to override `Update(elapsedTime)` in `SpecialTracksideCamera` to select the special locations if they're found, and only call `base.Update(elapsedTime)` if it can't find any special locations.

Reply:
Hi James, may I make a proposal? I have further added the roadcar based camera to this special points camera (see video mentioned in the ET forum), so I'd like to add it also within this pull request and then cope with your points. Else I have to make twice the adaptations accordingly to your requests. Can you agree?

Reply:
That's absolutely fine!

Reply:
I have checked again if this is possible, but comparing the two versions of Cameras.cs I see that the changes are in many points, not only in the central part trying to find a special point. This was the reason why I did it that way (which is not the best one, I know). The other possibility is to have two separate Update(elapsedTime) methods, which is paid with the duplication of some code lines.

Reply:
So I'm waiting for your comment about what solution you prefer.

Reply:
Sorry, but we should separate this code. I don't think you need to duplicate much/anything, because the `SpecialTracksideCamera` can call code in the `TracksideCamera`. If you need some code in both classes even with the `base.Update(elapsedTime)` call, put it into a `protected` method in `TracksideCamera` and you can call it from both classes as needed.

Reply:
I've generated an updated commit. I expect it is OK now for you.